src/site/content/en/blog/generic-sensor/index.md

```diff
@@ -0,0 +1,545 @@
+  activities in your application.
+- Sensor data models and coordinate systems are clearly defined, allowing browser vendors to
+  implement interoperable solutions.
+- The Generic Sensor based interfaces are not bound to the DOM (meaning they are neither `navigator`
+  nor `window` objects), and this opens up future opportunities for using the API within service
+  workers or implementing it in headless JavaScript runtimes, such as embedded
+  devices.
+- [Security and privacy](#privacy-and-security) aspects are the top priority for the Generic Sensor
+  API and provide much better security as compared to older sensor APIs. There is integration with
+  the Permissions API.
+- Automatic [synchronization with screen coordinates](#synchronization-with-screen-coordinates) is
+  available for `Accelerometer`, `Gyroscope`, `LinearAccelerationSensor`,
+  `AbsoluteOrientationSensor`, `RelativeOrientationSensor` and `Magnetometer`.
+
+## Browser compatibility
+
+The Generic Sensor API is supported by Google Chrome as of version 67. Most Chromium-derived
+browsers like Microsoft Edge, Opera, or Samsung Internet support this API as well. For other
+browsers, see [Can I use](https://caniuse.com/mdn-api_sensor). Note that the Generic Sensor API can
+be [polyfilled](#polyfill).
+
+## Available generic sensor APIs {: #available-generic-sensor-apis }
+
+At the time of writing, there are several sensors that you can experiment with.
+
+**Motion sensors:**
+
+- `Accelerometer`
+- `Gyroscope`
+- `LinearAccelerationSensor`
+- `AbsoluteOrientationSensor`
+- `RelativeOrientationSensor`
+- `GravitySensor`
+
+**Environmental sensors:**
+
+- `AmbientLightSensor`
+- `Magnetometer`
+
+## Feature detection {: #feature-detection }
+
+Feature detection of hardware APIs is tricky, since you need to detect both whether the browser
+supports the interface in question, _and_ whether the device has the corresponding sensor. Checking
+whether the browser supports an interface is straightforward. (Replace `Accelerometer` with any of
+the other interfaces mentioned [above](#available-generic-sensor-apis).)
+
+```js
+if ('Accelerometer' in window) {
+  // The `Accelerometer` interface is supported by the browser.
+  // Does the device have an accelerometer, though?
+}
+```
+
+For an actually meaningful feature detection result, you need to try to connect to the sensor, too.
+This example illustrates how to do that.
+
+```js
+let accelerometer = null;
+try {
+  accelerometer = new Accelerometer({ frequency: 10 });
+  accelerometer.addEventListener('error', (event) => {
+    // Handle runtime errors.
+    if (event.error.name === 'NotAllowedError') {
+      console.log('Permission to access sensor was denied.');
+    } else if (event.error.name === 'NotReadableError') {
+      console.log('Cannot connect to the sensor.');
+    }
+  });
+  accelerometer.addEventListener('reading', () => reloadOnShake(accelerometer));
```

Comment:
Some users may be confused with this function `reloadOnShake()`.
```suggestion
  accelerometer.addEventListener('reading', () => () => { /* Handle reading */ });
```

Reply:
Yes, good catch. The function doesn't make sense in the generalization.